CONTRIBUTING.md

```diff
@@ -40,177 +264,782 @@ since someone else can often do this if you include a complete example. We
+Additional documentation can be found in
+[`dev/doc/build-system.dune.md`][dev-doc-dune], and in [the official
+Dune documentation][dune-doc].
+
+The legacy make-based system is still available.  If you wish to use
+it, you need to start by running `./configure -profile devel`.  Most
+of the available targets are not documented, so you will need to ask.
+
+#### Continuous integration ####
+
+Continuous integration (CI) testing is key in ensuring that the
+`master` branch is kept in a well-functioning state at all times, and
+that no accidental compatibility breakages are introduced.  Our CI is
+quite extensive since it includes testing many external projects, some
+of them taking more than an hour to compile.  However, you can get
+partial results much more quickly (when our CI is not overloaded).
+
+The main documentation resources on our CI are:
+
+- the [README for users, i.e. plugin and library authors][CI-README-users];
+- the [README for developers, and contributors][CI-README-developers];
+- the README of the [user-overlays][] directory.
+
+Preparing an overlay is a step that everyone goes through at some
+point.  All you need to know to prepare an overlay manually is in the
+README in the [user-overlays][] directory.  You might
+want to use some additional tooling such as the `make ci-*` targets of
+`Makefile.ci`, the Nix support for getting the dependencies of the
+external projects (see the README in [`dev/ci/nix`](dev-ci-nix), and
+the (undocumented so far) `dev/tools/create_overlays.sh` script.
+
+More work is to be done on understanding how each developer proceeds
+to prepare overlays, and propose a simplified and documented
+procedure.
+
+We also have a benchmarking infrastructure, which is documented [on
+the wiki][jenkins-doc].
+
+#### Code owners, issue and pull request templates ####
+
+These files can be found in the [`.github`](.github) directory.  The
+templates are particularly useful to remind contributors what
+information we need for them, and, in the case of PRs, to update the
+documentation, changelog, and test-suite when relevant.
+
+GitHub now supports setting up multiple issue templates, and we could
+use this to define distinct requirements for various kind of bugs,
+enhancement and feature requests.
+
+#### Style guide ####
+
+There exists an [old style guide][old-style-guide] whose content is
+still mostly relevant.  We don't use a code formatter at the current
+time, and we are reluctant to formatting-only commits, or commits
+formatting parts of code that are unchanged beyond formatting.
+However, it is still a good idea if you don't know how to format a
+block to use the formatting that [ocamlformat][] would give
+
+#### OCaml resources ####
+
+You may find lots of OCaml resources on <http://ocaml.org/>, including
+documentation, a Discourse forum, the package archive, etc.  You may
+also want to refer to the [Dune documentation][dune-doc].
+
+#### Git documentation, tips and tricks ####
+
```

Comment:
This section has very little to do with Coq or its dev process.
I would not surprised to find a github tutorial doing the same: https://help.github.com/en/articles/about-pull-requests

Reply:
FTR, I think it is important to talk about the two-remote setup, and to start from a fresh master, and I don't know any tutorial that does it. The way it is written is specific to Coq only because of the URLs it uses. Anyway, I didn't want this part to make the reading more difficult for contributors who knew already about that kind of stuff, so this is why I put it in a resources section, and it has a title containing "tips and tricks".